### ISSUE TITLE
Option to use Substrate instead of Substitute

### ISSUE THREAD

**Issue by rctfan929** [pwn20wndstuff/Undecimus#2005]
I am on a 6s would like the option to use Substrate. Possibly you could add the chooser that was in u0 4.3.1 back.

I have already tried uninstalling Substitute and installing Substrate through Cydia. This causes my tweaks to stop working.

**Comment by SarKaa** [comment]
I agree, I am on checkra1n, and want to keep unc0ver signed as a backup when I don’t have access to a computer, but everytime I do use it, I need to manually remove substitute, and it would be easier to use substrate by default

**Comment by badger200** [comment]
> I agree, I am on checkra1n, and want to keep unc0ver signed as a backup when I don’t have access to a computer, but everytime I do use it, I need to manually remove substitute, and it would be easier to use substrate by default

How are you able to remove Substitute? I am having massive problems due to v5 installing Substitute and not being able to revert to Substrate. Running 4.3.1 it dies trying to convert to Substrate claiming I need to run “dpkg —configure -a” which does nothing (no output) and doesn’t solve it.

If I try to remove Substitute like a package in Cydia, it acts like it’s going to remove ALL my tweaks, which I don’t want to do.

**Comment by SarKaa** [comment]
> > I agree, I am on checkra1n, and want to keep unc0ver signed as a backup when I don’t have access to a computer, but everytime I do use it, I need to manually remove substitute, and it would be easier to use substrate by default
> 
> How are you able to remove Substitute? I am having massive problems due to v5 installing Substitute and not being able to revert to Substrate. Running 4.3.1 it dies trying to convert to Substrate claiming I need to run “dpkg —configure -a” which does nothing (no output) and doesn’t solve it.
> 
> If I try to remove Substitute like a package in Cydia, it acts like it’s going to remove ALL my tweaks, which I don’t want to do.

Tbf I used a snap back backup, or snapshot, so I didnt have to manually remove anything

**Comment by SarKaa** [comment]
Up

> > I agree, I am on checkra1n, and want to keep unc0ver signed as a backup when I don’t have access to a computer, but everytime I do use it, I need to manually remove substitute, and it would be easier to use substrate by default
> 
> How are you able to remove Substitute? I am having massive problems due to v5 installing Substitute and not being able to revert to Substrate. Running 4.3.1 it dies trying to convert to Substrate claiming I need to run “dpkg —configure -a” which does nothing (no output) and doesn’t solve it.
> 
> If I try to remove Substitute like a package in Cydia, it acts like it’s going to remove ALL my tweaks, which I don’t want to do.

Maybe try reinstalling substrate, that should automatically remove substitute without removing tweaks

**Comment by bigrck64** [comment]
I would be nice to have the choice between substrate & substitute, true.

**Comment by eglacias** [comment]
Yes this must be a known issue, I get exactly the same problem on an iPhone 6s Plus on iOS 12.4.1

And I cannot remove substitute because if same error and install substrate to get checkrain working.
either check rain needs to start using substitute or uncover needs to use substrate. 

 We really like to have uncover as on device back up jailbreak. For now I have to exclusively use uncover  which I would prefer not as switching back to check rain now requires Internet connection, restore rootFS, and reload with batchomatic.

**Comment by eglacias** [comment]
And reinstalling substrate doesn’t work. That’s what I used to have to do to get checkrain up but I’m guessing uncover is working so well and rewritten  very different right now

**Comment by SarKaa** [comment]
See this is what I mean, susbtitute causes errors with incompatible tweaks, but it does have some benefits, so I think it would be good to give the user a choice. This way, we can continue using checkra1n as our main jailbreak, but also keep unc0ver installed as a backup

**Comment by liuxuan30** [comment]
I think when reinstall substrate from cydia, the next time you reboot your device, it will revert back to substitute again. So it's really painful not able to choose substrate for Unc0ver. Is there any specific reason Unc0ver v5 don't support substrate any more? I get the idea that arm64e won't work with substrate, but at least for older devices, let us have the option to choose. 

Thanks!
